interfaces/inc/c_pal/threadpool.h

```diff
@@ -40,13 +43,11 @@ MOCKABLE_FUNCTION(, void, threadpool_destroy_work_item, THANDLE(THREADPOOL), thr
 
 MOCKABLE_FUNCTION(, int, threadpool_schedule_work, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context);
 
-MOCKABLE_FUNCTION(, int, threadpool_timer_start, THANDLE(THREADPOOL), threadpool, uint32_t, start_delay_ms, uint32_t, timer_period_ms, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context, TIMER_INSTANCE_HANDLE*, timer_handle);
-
-MOCKABLE_FUNCTION(, int, threadpool_timer_restart, TIMER_INSTANCE_HANDLE, timer, uint32_t, start_delay_ms, uint32_t, timer_period_ms);
+MOCKABLE_FUNCTION(, int, threadpool_timer_start, THANDLE(THREADPOOL), threadpool, uint32_t, start_delay_ms, uint32_t, timer_period_ms, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context, THANDLE(TIMER_INSTANCE)*, timer_handle);
```

Comment:
>THANDLE(TIMER_INSTANCE)* [](http://example.com/codeflow?start=206&length=24)

Reading this I really had to wonder why we're not returning THANDLE(TIMER_INSTANCE).
But this is just a remark, I am not saying to rev the API at the same time. It's probably not a good idea to do that as it will ripple quite a lot.

Reply:
only reason is to keep the API signature not changed, will see if change to return THANDLE(TIMER) is needed after propagate this change is done